File: String/Java/wildCardPattern.java
public class wildCardPattern {
    public static boolean isMatch(String text, String pattern) {
        if (pattern.isEmpty()) {
            return text.isEmpty();
        }
        if (text.isEmpty() && pattern.charAt(0) != '*') {
            return false;
        }
        if (pattern.charAt(0) == '*') {
            return isMatch(text, pattern.substring(1)) || (!text.isEmpty() && isMatch(text.substring(1), pattern));
        } else {
            return !text.isEmpty() && (pattern.charAt(0) == '?' || pattern.charAt(0) == text.charAt(0)) && isMatch(text.substring(1), pattern.substring(1));
        }
    }

    public static void main(String[] args) {
        String text = "baaabab";
        String pattern = "*****ba*****ab";
        System.out.println(isMatch(text, pattern) ? "true" : "false");
    }
}

File: String/C/wildCardPattern.c
#include <stdio.h>
#include <stdbool.h>
#include <string.h>

bool isMatch(const char *text, const char *pattern) {
    if (*pattern == '\0') {
        return *text == '\0';
    }
    if (*text == '\0' && *pattern != '*') {
        return false;
    }
    if (*pattern == '*') {
        return isMatch(text, pattern + 1) || (*text != '\0' && isMatch(text + 1, pattern));
    } else {
        return *text != '\0' && (*pattern == '?' || *pattern == *text) && isMatch(text + 1, pattern + 1);
    }
}

int main() {
    const char *text = "baaabab";
    const char *pattern = "*****ba*****ab";
    printf("%s\n", isMatch(text, pattern) ? "true" : "false");
    return 0;
}